app/src/main/java/com/rajatgoyal/puzzle15/ui/GameActivity.java

```diff
@@ -56,20 +58,14 @@ public void run() {
             time /= 60;
             hours = (int) time % 24;
 
-            timerTextView.setText(new Time((int) (currTime / 1000)).toString());
+            timerTextView.setText(new Time(hours, minutes, seconds).toString());
```

Comment:
We can use `Time(currTime)` here now.

Reply:
This is fine. Since we are calculating hours, minutes and seconds here, there is no need for them to be recalculated in Time class too.

Reply:
I meant that we can remove the hours, minutes and seconds calculation from here and do it in the Time model only.

Reply:
It will be handled when #29 is solved